ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs

```diff
@@ -0,0 +1,37 @@
+
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE TypeSynonymInstances  #-}
+
+module Ouroboros.Consensus.Protocol (
```

Comment:
Come to think of it, this is similar to my comment over on `TriggerHardFork`: it's a convenience offering, not necessary to Consensus's own definitions. (Hmm, if enough of these accumulate, perhaps we should eventually offer a separate package containing them.)

Actually, I'm not sure I see the utility of defining this at all. What would the effect on downstream code be if we just dropped the `Protocol` GADT instead of refactoring it? I imagine you've had to consider that in the downstream PRs you've opened -- I haven't looked at those. I did just now look at `cardano-node origin/master` to see how it uses the existing GADT, and I'm fairly confident we could change their `SomeConsensusProtocol` constructor to just carry your new `BlockType blk` GADT and a `ProtocolInfo`:

```
data SomeConsensusProtocol where

     SomeConsensusProtocol :: SomeConsensusProtocolConstraints blk
                           => BlockType blk
                           -> ProtocolInfo m blk
                           -> SomeConsensusProtocol
```

Ultimately, all the unwrappings of `SomeConsensusProtocol` I saw only used it as the universe tag (your type) and/or immediately converted it to a `ProtocolInfo`. .... maybe I missed a log message or some such that dumped the old `Protocol` GADT's constructor arguments?

Less is more, etc. What do you think?

-----

If we don't manage to just simplify by removing this, then I'll make the following requests:

  * Would you summarize that above sentiment (about convenience offering and whatever motivation we come up with for keeping this) as a Haddock comment on `Protocol` below?

  * Would you also rename the two classes to `ToProtocolInfo` and `ToProtocolClientInfo`? And probably the methods to `toProtcolInfo` and `toProtocolClientInfo`. I want to avoid potential confusion with the `ConsensusProtocol` class and the existing `Protocol` class in `typed-protocols`.
  
  * As for `TriggerHardFork`, I'd like to rename this module. See my comment there for the underlying reasoning. Maybe this should be `Ouroboros.Consensus.ProtocolInfo.ToProtocolInfo`. Thoughts?